### PR TITLE
TextIO/CompressedSource: support splitting AUTO mode files into bundles

### PR DESCRIPTION
**Pull request by dhalperi**
This is a backport of apache/incubator-beam#980

R: @jkff please review for backporting errors

### CODE REVIEW

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/io/CompressedSource.java [hunk 1]
```diff
@@ -284,7 +284,7 @@ public void validate() {
    * from the requested file name that the file is not compressed.
    */
   @Override
-  protected final boolean isSplittable() throws Exception {
+  protected final boolean isSplittable() {
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/CompressedSource.java]: Problem: this is a backwards-incompatible change I have to fix in DataflowJavaSDK

Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/CompressedSource.java]: Fixed

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/TextIOTest.java [hunk 2]
```diff
@@ -144,8 +172,63 @@ private TestDataflowPipelineOptions buildTestPipelineOptions() {
...
+    tinyGz = writeToFile(TINY, "tiny.gz", GZIP);
+    tinyBzip2 = writeToFile(TINY, "tiny.bz2", BZIP2);
+    // large files
+    largeTxt = writeToFile(LARGE, "large.txt", UNCOMPRESSED);
+    largeGz = writeToFile(LARGE, "large.gz", GZIP);
+    largeBzip2 = writeToFile(LARGE, "large.bz2", BZIP2);
+  }
+
+  @AfterClass
+  public static void testdownClass() throws IOException {
```
Comment (jkff) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/TextIOTest.java]: Didn't notice last time but this should be called teardownClass

**Comment by dhalperi** [comment]
Thanks! Will fix the name in both places

**Pull request merged**